wgpu-hal/src/gles/device.rs

```diff
@@ -391,7 +443,26 @@ impl crate::Device<super::Api> for super::Device {
     unsafe fn unmap_buffer(&self, buffer: &super::Buffer) -> Result<(), crate::DeviceError> {
         let gl = &self.shared.context.lock();
         gl.bind_buffer(buffer.target, Some(buffer.raw));
-        gl.unmap_buffer(buffer.target);
+
+        if self
```

Comment:
may just check `emulate_map_allocation.take()` and go from there instead of checking the private caps